pkg/config/types/v1alpha1/cluster_config.go

```diff
@@ -143,212 +117,25 @@ func (e *EtcdConfig) CA() *x509.PEMEncodedCertificateAndKey {
-	}
-
-	kubeletConfig := &kubeletconfig.KubeletConfiguration{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "KubeletConfiguration",
-			APIVersion: "kubelet.config.k8s.io/v1beta1",
-		},
-		FeatureGates: map[string]bool{},
-	}
-
-	proxyConfig := &kubeproxyconfig.KubeProxyConfiguration{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "KubeProxyConfiguration",
-			APIVersion: "kubeproxy.config.k8s.io/v1alpha1",
-		},
-		Mode: "ipvs",
-		IPVS: kubeproxyconfig.KubeProxyIPVSConfiguration{
-			Scheduler: "lc",
-		},
-	}
-
-	// TODO(andrewrynhard): This should only be configured when the runtime mode
-	// is container.
-	f := false
-	kubeletConfig.FailSwapOn = &f
-	// See https://github.com/kubernetes/kubernetes/issues/58610#issuecomment-359552443
-	maxPerCore := int32(0)
-	proxyConfig.Conntrack.MaxPerCore = &maxPerCore
-
-	if err := kubeletconfigv1beta1scheme.AddToScheme(kubeadmscheme.Scheme); err != nil {
-		return "", err
-	}
-
-	if err := kubeproxyconfigv1alpha1scheme.AddToScheme(kubeadmscheme.Scheme); err != nil {
-		return "", err
-	}
-
-	encodedObjs := [][]byte{}
-	for _, obj := range []kuberuntime.Object{initConfig, clusterConfig} {
-		encoded, err := kubeadmutil.MarshalToYamlForCodecs(obj, kubeadmv1beta2.SchemeGroupVersion, kubeadmscheme.Codecs)
-		if err != nil {
-			return "", err
-		}
-		encodedObjs = append(encodedObjs, encoded)
-	}
-
-	encoded, err := kubeadmutil.MarshalToYamlForCodecs(kubeletConfig, kubeletconfigv1beta1scheme.SchemeGroupVersion, kubeadmscheme.Codecs)
-	if err != nil {
-		return "", err
-	}
-	encodedObjs = append(encodedObjs, encoded)
-
-	encoded, err = kubeadmutil.MarshalToYamlForCodecs(proxyConfig, kubeproxyconfigv1alpha1scheme.SchemeGroupVersion, kubeadmscheme.Codecs)
-	if err != nil {
-		return "", err
-	}
-	encodedObjs = append(encodedObjs, encoded)
-
-	kubeadmConfig := bytes.Join(encodedObjs, []byte("---\n"))
-
-	return string(kubeadmConfig), nil
+// Token implements the Configurator interface.
+func (c *ClusterConfig) Token() cluster.Token {
+	return c
 }
 
-func (c *ClusterConfig) controlplaneConfig() (string, error) {
-	controlPlaneConfig := &kubeadm.JoinConfiguration{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "JoinConfiguration",
-			APIVersion: "kubeadm.k8s.io/v1beta2",
-		},
-		ControlPlane: &kubeadm.JoinControlPlane{
-			CertificateKey: c.CertificateKey,
-		},
-		Discovery: kubeadm.Discovery{
-			BootstrapToken: &kubeadm.BootstrapTokenDiscovery{
-				Token:                    c.Token,
-				APIServerEndpoint:        c.ControlPlane.IPs[0] + ":443",
-				UnsafeSkipCAVerification: true,
-			},
-		},
-		NodeRegistration: kubeadm.NodeRegistrationOptions{
-			CRISocket: constants.ContainerdAddress,
-			// KubeletExtraArgs: ,
-		},
-	}
-
-	encodedObjs := [][]byte{}
-	encoded, err := kubeadmutil.MarshalToYamlForCodecs(controlPlaneConfig, kubeadmv1beta2.SchemeGroupVersion, kubeadmscheme.Codecs)
-	if err != nil {
-		return "", err
+// ID implements the Configurator interface.
+func (c *ClusterConfig) ID() string {
```

Comment:
It seems very odd to have `ClusterConfig.ID()` return the cluster's node bootstrap ID

Reply:
This uses the nomenclature from Kubernetes. A bootstrap token has and id and a secret portion. Separated by the `.`.

Reply:
Right, I'm talking about the namespace:  bootstrap token aspects do not seem to be well-situated in the root of cluster config.  Put another way, one would expect `ClusterConfig.ID()` would return something like an ID of the _cluster_, not the bootstrap token _of_ the cluster.

Reply:
ahh yes, this was just a struct where we could satisfy the interface.